server/src/services/library.service.ts

```diff
@@ -540,6 +533,34 @@ export class LibraryService implements OnEvents {
     return JobStatus.SUCCESS;
   }
 
+  // Checks if an online asset should be marked as offline, either due to missing path, or path outside of any import path
+  async handleOfflineCheck(job: ILibraryOfflineJob): Promise<JobStatus> {
+    const asset = await this.assetRepository.getById(job.id);
+
+    if (!asset || asset.isOffline) {
+      // We only care about online assets, we exit here if offline
+      return JobStatus.SKIPPED;
+    }
+
+    const exists = await this.storageRepository.checkFileExists(asset.originalPath, R_OK);
+
+    if (exists) {
+      const isInPath = job.importPaths.find((path) => asset.originalPath.startsWith(path));
```

Comment:
this condition can actually be checked outside of this if statement. I think it would be more clear if the code was like this:
```ts
if(!fileExists || !validPath) { 
    this.logger.debug(`Marking asset as offline: ${asset.originalPath}`);
    await this.assetRepository.update({ id: asset.id, isOffline: true });
}

return JobStatus.SUCCESS

Reply:
I've reworked the logic of this method and it should be much clearer